### PR TITLE
Revert "equivalence class hash calculation in scheduler"

### PR DESCRIPTION
**Pull request by bsalamat**
Reverts kubernetes/kubernetes#60753

ref/ #61512
/sig scheduling

Let's wait and see how the tests go.

```release-note
NONE
```

### CODE REVIEW

**Comment by resouer** [comment]
/assign

I will take a very quick look

File: pkg/scheduler/core/generic_scheduler.go [hunk 1]
```diff
@@ -490,64 +494,46 @@ func podFitsOnNode(
...
-								res.FailReasons = append(res.FailReasons, reasons...)
-								predicateResults[predicateKey] = res
-							} else {
-								predicateResults[predicateKey] = HostPredicate{Fit: fit, FailReasons: reasons}
-							}
-							result := predicateResults[predicateKey]
-							ecache.UpdateCachedPredicateItem(
-								pod.GetName(), info.Node().GetName(),
-								predicateKey, result.Fit, result.FailReasons, equivCacheInfo.hash, false)
+				if !eCacheAvailable || invalid {
```
Comment (resouer) [inline pkg/scheduler/core/generic_scheduler.go]: should use inline func to fix the previous deadlock issue here

**Comment by resouer** [comment]
Others looks good except the deadlock fix should not be reverted.

**Comment by liggitt** [comment]
> Others looks good except the deadlock fix should not be reverted.

can we merge this to fix tests and do the deadlock fix separately?

**Comment by resouer** [comment]
@liggitt Sure, I can send out a seperate PR to fix the deadlock.

**Comment by resouer** [comment]
@bsalamat Please set release note :)

**Pull request merged**